cmd/deployIntegrationArtifact_test.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	cpi "github.com/SAP/jenkins-library/pkg/cpi"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type deployIntegrationArtifactMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newDeployIntegrationArtifactTestsUtils() deployIntegrationArtifactMockUtils {
+	utils := deployIntegrationArtifactMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunDeployIntegrationArtifact(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Positive"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.NoError(t, err)
+	})
+
+	t.Run("Failed case of Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Negative"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.Error(t, err)
```

Comment:
I would like to suggest to use `EqualError` in order to assert that the error we got is the expected error. Otherwise we might got an error, but a different one ...

Reply:
done